src/main/java/org/rutebanken/tiamat/versioning/ValidityUpdater.java

```diff
@@ -37,34 +49,46 @@
+     * @param newVersion      the version to update valid between on
      * @param defaultFromTime
      * @return the new version's valid from date
      */
     protected <T extends EntityInVersionStructure> Instant updateValidBetween(T existingVersion, T newVersion, Instant defaultFromTime) {
 
         instantiateValidBetween(newVersion);
 
-        if(newVersion.getValidBetween().getFromDate() != null
+        if (newVersion.getValidBetween().getFromDate() != null
                 && newVersion.getValidBetween().getToDate() != null
                 && newVersion.getValidBetween().getFromDate().isAfter(newVersion.getValidBetween().getToDate())) {
-            throw new IllegalArgumentException("Entity " + newVersion.getNetexId() + " has from date " + newVersion.getValidBetween().getFromDate() + " after to date " + newVersion.getValidBetween().getToDate());
+            throw new IllegalArgumentException(messages.get(VALIDATION_FROM_DATE_AFTER_TO_DATE, newVersion.getNetexId(), formatToLocalDateTime(newVersion.getValidBetween().getFromDate()), formatToLocalDateTime(newVersion.getValidBetween().getToDate())));
         }
 
-        if(newVersion.getValidBetween().getFromDate() == null) {
-            logger.info("From date not set for new version of entity with ID: {}. Setting default from time: {}", newVersion.getNetexId(), defaultFromTime);
+        if (newVersion.getValidBetween().getFromDate() == null) {
+            logger.info(messages.get(VALIDATION_FROM_DATE_NOT_SET, newVersion.getNetexId(), formatToLocalDateTime(defaultFromTime)));
             newVersion.getValidBetween().setFromDate(defaultFromTime);
         }
 
-        if(existingVersion != null && existingVersion.getValidBetween() != null) {
-            String entityString = existingVersion.getNetexId() + " " +existingVersion.getVersion();
-            validateNewVersionDateAfter("Existing version " + entityString + " to date", existingVersion.getValidBetween().getToDate(), newVersion.getValidBetween().getFromDate());
-            validateNewVersionDateAfter("Existing version " + entityString + " from date", existingVersion.getValidBetween().getFromDate(), newVersion.getValidBetween().getFromDate());
+        if (existingVersion != null && existingVersion.getValidBetween() != null) {
+            String entityString = existingVersion.getNetexId() + " " + existingVersion.getVersion();
+            validateNewVersionDateAfter(VALIDATION_TO_DATE_AFTER_NEXT_VERSION_FROM_DATE, entityString, existingVersion.getValidBetween().getToDate(), newVersion.getValidBetween().getFromDate());
+            validateNewVersionDateAfter(VALIDATION_FROM_DATE_AFTER_NEXT_VERSION_FROM_DATE, entityString, existingVersion.getValidBetween().getFromDate(), newVersion.getValidBetween().getFromDate());
         }
 
         return newVersion.getValidBetween().getFromDate();
     }
 
+    /**
+     * FIXME : any way to have incoming TZ instead of relying on default TZ ?
```

Comment:
ExportTimeZone is only used for dates being exported or serialized in messages like this, but shall never be used for parsing incoming data (hence, the name). Dates in incoming graphql requests are parsed in DateScalar. This parser requires timezone in the incoming datetime (not using export time zone). Therefore, the timezone is known, and could maybe be used to generate the exception message? Anyhow, the exception message will at the time of writing contain the correct datetime, but maybe represented in another timezone.